src/org/camunda/latera/bss/connectors/hid/hydra/Subscription.groovy

```diff
@@ -110,10 +112,48 @@ trait Subscription {
       endDate            : null,
       chargeLogEndDate   : null,
       evaluateDiscounts  : true
-    ], input)
+    ]
 
-    def unitId = getGoodUnitId(params.goodId)
+    LinkedHashMap existingSubscription = [:]
+    if (notEmpty(input.subscriptionId)) {
+      LinkedHashMap subscription = getSubscription(input.subscriptionId)
+      existingSubscription += [
+        subscriptionId     : subscription.n_subscription_id,
+        customerId         : subscription.n_customer_id,
+        accountId          : subscription.n_account_id,
+        docId              : subscription.n_doc_id,
+        goodId             : subscription.n_service_id,
+        equipmentId        : subscription.n_object_id,
+        parSubscriptionId  : subscription.n_par_subscription_id,
+        prevSubscriptionId : subscription.n_prev_subscription_id,
+        quant              : subscription.n_quant,
+        payDay             : subscription.n_pay_day,
+        beginDate          : subscription.d_begin,
+        endDate            : subscription.d_end,
+        chargeLogEndDate   : subscription.d_charge_log_end
+      ]
+    }
 
+    if (notEmpty(input.parSubscriptionId)) {
+      LinkedHashMap parentSubscription = getSubscription(input.parSubscriptionId)
```

Comment:
looks like it is not used

Reply:
What is not used?

Reply:
usually you don't need variable ```parentSubscription``` in such situations, in ruby it's something like
```
existingSubscription = existingSubscription.merge(getSubscription(input.parSubscriptionId).slice(:n_customer_id, :n_account_id, :n_doc_id, :n_object_id)) 
```

don't you have groovy way for this?

Reply:
There is [subMap|http://docs.groovy-lang.org/latest/html/groovy-jdk/java/util/Map.html#subMap(java.lang.Object[])] method in groovy which can be used for this.

But input arguments of updateSubscription have different names from output of SELECT - compare `subscriptionId` and `n_subscription_id`. And there is no way easier than currently one to convert them properly.